x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/AuthenticationServiceTests.java

```diff
@@ -331,10 +342,20 @@ public void testAuthenticateBothSupportSecondSucceeds() throws Exception {
         } else {
             when(secondRealm.token(threadContext)).thenReturn(token);
         }
-        final String reqId = AuditUtil.getOrGenerateRequestId(threadContext);
+        boolean requestIdAlreadyPresent = randomBoolean();
+        SetOnce<String> reqId = new SetOnce<>();
+        if (requestIdAlreadyPresent) {
+            reqId.set(AuditUtil.getOrGenerateRequestId(threadContext));
+            assertThat(reqId.get(), not(nullValue()));
```

Comment:
Nit: this assertion of reqId.get() being non-null does not exist in other places. Do we need it here?